src/main.py

```diff
@@ -1,21 +1,34 @@
 import sys
+from argparse import ArgumentParser
 
 from command_line import enhanced_prompt, reduced_prompt, CommandHandler
 from version import VERSION_STRING
 
-
-COMMAND_MODE = "--command-line" in sys.argv or "-c" in sys.argv
-REDUCED_MODE = "--reduced" in sys.argv or "-r" in sys.argv
-
-
 if __name__ == "__main__":
-    if COMMAND_MODE:
+    parser = ArgumentParser(
+        description="A new Minecraft world editor that aims to be flexible, "
+        "extendable and support most editions of Minecraft"
+    )
+    parser.add_argument(
+        "-c",
+        "--command-line",
+        action="store_true",
+        help="use the commandline interface",
+    )
+    parser.add_argument(
+        "-r",
+        "--reduced",
+        action="store_true",
+        help="use a reduced prompt (commandline mode only)",
+    )
+    args = parser.parse_args()
+    if args.command_line:
```

Comment:
Like Podshot said, the Reduced option doesn't get inside this if statement

Reply:
Maybe we want to give some kind of warning if reduced is used without the command_line option

Reply:
Yes. I noticed that too. The current behaviour is exactly as it was before.
Which solution is the better one? Activating commandline mode when only --reduced is passed or complaining if only --reduced it used?

Reply:
I vote for complaining but would love to hear Podshot's response too

Reply:
I would vote for complaining, since while the old behavior was to just error, it would make more sense to inform the user that they need to include the `--command-line` option

Reply:
Is this ok?